frontend/src/stimulus/controllers/dynamic/work-packages/activities-tab/index.controller.ts

```diff
@@ -0,0 +1,422 @@
+      const text = await response.text();
+      Turbo.renderStreamMessage(text);
+      this.setLastUpdateTimestamp();
+      setTimeout(() => {
+        if (this.sortingValue === 'asc' && journalsContainerAtBottom) {
+          // scroll to (new) bottom if sorting is ascending and journals container was already at bottom before a new activity was added
+          if (this.isMobile()) {
+            this.scrollInputContainerIntoView(300);
+          } else {
+            this.scrollJournalContainer(this.journalsContainerTarget, true);
+          }
+        }
+      }, 100);
+    }
+  }
+
+  private rescueEditorContent() {
+    const ckEditorInstance = this.getCkEditorInstance();
+    if (ckEditorInstance) {
+      const data = ckEditorInstance.getData({ trim: false });
+      if (data.length > 0) {
+        localStorage.setItem(this.localStorageKey, data);
+      }
+    }
+  }
+
+  private populateRescuedEditorContent() {
+    const rescuedEditorContent = localStorage.getItem(this.localStorageKey);
+    if (rescuedEditorContent) {
+      this.openEditorWithInitialData(rescuedEditorContent);
+      localStorage.removeItem(this.localStorageKey);
+    }
+  }
+
+  private handleInitialScroll() {
+    if (window.location.hash.includes('#activity-')) {
+      this.scrollToActivity();
+    } else if (this.sortingValue === 'asc') {
+      this.scrollToBottom();
+    }
+  }
+
+  private scrollToActivity() {
+    const activityId = window.location.hash.replace('#activity-', '');
+    const activityElement = document.getElementById(`activity-${activityId}`);
+    activityElement?.scrollIntoView({ behavior: 'smooth' });
+  }
+
+  private scrollToBottom() {
+    const scrollableContainer = jQuery(this.element).scrollParent()[0];
+    if (scrollableContainer) {
+      setTimeout(() => {
+        scrollableContainer.scrollTop = scrollableContainer.scrollHeight;
+      }, 400);
+    }
+  }
+
+  setFilterToOnlyComments() { this.filterValue = 'only_comments'; }
+  setFilterToOnlyChanges() { this.filterValue = 'only_changes'; }
+  unsetFilter() { this.filterValue = ''; }
+
+  private getCkEditorInstance():ICKEditorInstance | null {
+    const AngularCkEditorElement = this.element.querySelector('opce-ckeditor-augmented-textarea');
+    return AngularCkEditorElement ? jQuery(AngularCkEditorElement).data('editor') as ICKEditorInstance : null;
+  }
+
+  private getInputContainer():HTMLElement | null {
+    return this.element.querySelector('#input-container');
+  }
+
+  // TODO: get rid of static width value and reach for a more CSS based solution
+  private isMobile():boolean {
+    return window.innerWidth < 1279;
+  }
+
+  // TODO: get rid of static width value and reach for a more CSS based solution
+  private isSmViewPort():boolean {
+    return window.innerWidth < 543;
+  }
+
+  // TODO: get rid of static width value and reach for a more CSS based solution
+  private isMdViewPort():boolean {
+    return window.innerWidth >= 543 && window.innerWidth < 1279;
+  }
```

Comment:
Agreed, do not use fixed size values here and instead use our CSS breakpoints

Reply:
I hope, we can get rid of all of that when we have the standalone activity tab on mobile for the workpackage page as well. We then have the same constraints on all viewports, which would be great!